Modules/_decimal/_decimal.c

```diff
@@ -39,7 +39,9 @@
 
 #include "docstrings.h"
 
-typedef struct {
+struct PyDecContextObject;
```

Comment:
You only need this for the forward decl; there's no need for the added struct names.

Reply:
If we just keep this forward declaration and don't add a name to the structure, the compilation will fail.

I built it locally with:

```shell
./configure --without-decimal-contextvar
make
```

<details><summary>Error Message</summary>
<p>

```
...
./Modules/_decimal/_decimal.c: In function ‘context_dealloc’:
./Modules/_decimal/_decimal.c:1315:14: warning: comparison of distinct pointer types lacks a cast
 1315 |     if (self == state->cached_context) {
      |              ^~
./Modules/_decimal/_decimal.c: In function ‘current_context_from_dict’:
./Modules/_decimal/_decimal.c:1595:30: warning: assignment to ‘struct PyDecContextObject *’ from incompatible pointer type ‘PyDecContextObject *’ [-Wincompatible-pointer-types]
 1595 |     modstate->cached_context = (PyDecContextObject *)tl_context;
      |                              ^
./Modules/_decimal/_decimal.c:1596:29: error: invalid use of undefined type ‘struct PyDecContextObject’
 1596 |     modstate->cached_context->tstate = tstate;
      |                             ^~
./Modules/_decimal/_decimal.c: In function ‘current_context’:
./Modules/_decimal/_decimal.c:1608:55: error: invalid use of undefined type ‘struct PyDecContextObject’
 1608 |     if (state->cached_context && state->cached_context->tstate == tstate) {
      |                                                       ^~
```

</p>
</details> 

Unfortunately, CI will not fail because it doesn't add the compile option `--without-decimal-contextvar`. So the wrong statement is not caught by the compiler.

Reply:
Of course, I forgot you need the configure switch; thanks for the heads up!

Reply:
Still, you only need the added struct name for `PyDecContextObject`; there's no need to add names to the other structs.

Generally, for CPython PRs, it is good practice to always keep the diff of a PR as small as possible.

Reply:
Thanks! I will do it soon~